ui/helpers/utils/util.js

```diff
@@ -408,3 +408,19 @@ export function getURLHost(url) {
 export function getURLHostName(url) {
   return getURL(url)?.hostname || '';
 }
+
+// Once we reach this threshold, we switch to higher unit
+const MINUTE_CUTOFF = 90 * 60;
+const SECOND_CUTOFF = 90;
+
+export const toHumanReadableTime = (t, milliseconds) => {
```

Comment:
I don't think a test would require full localization support, since you're passing `t` directly to this function (and not getting it from a provider). What if you provided a fake? 

``` javascript
const t = (key, [number]) => {
  switch (key) {
    case 'gasTimingSecondsShort':
      return `${number} sec`;
    case 'gasTimingMinutesShort':
      return `${number} min`;
    case 'gasTimingHoursShort':
      return `${number} hrs`;
  }
}

toHumanReadableTime(t, 120_000);
```